content/graph/DFSMatching.h

```diff
@@ -5,37 +5,35 @@
  * Source:
  * Description: This is a simple matching algorithm but should
  * be just fine in most cases. Graph $g$ should be a list of
- * neighbours of the left partition. $n$ is the size of the left
- * partition and $m$ is the size of the right partition.
- * If you want to get the matched pairs, $match[i]$ contains
+ * neighbours of the left partition, and $btoa$ should be
+ * a vector with size equal to the size of the right partition.
+ * If you want to get the matched pairs, $btoa[i]$ contains
  * match for vertex $i$ on the right side or $-1$ if it's not
  * matched.
  * Time: O(EV) where $E$ is the number of edges and V is the number of vertices.
+ * Usage: vi ba(m, -1); dfsMatching(g, ba);
  * Status: works
  */
 #pragma once
 
-vi match;
-vector<bool> seen;
-bool find(int j, const vector<vi>& g) {
-	if (match[j] == -1) return 1;
-	seen[j] = 1; int di = match[j];
+bool find(int j, vector<vi>& g, vi& btoa, vector<bool>& vis) {
+	if (btoa[j] == -1) return 1;
+	vis[j] = 1; int di = btoa[j];
 	trav(e, g[di])
-		if (!seen[e] && find(e, g)) {
-			match[e] = di;
+		if (!vis[e] && find(e, g, btoa, vis)) {
+			btoa[e] = di;
 			return 1;
 		}
 	return 0;
 }
-int dfs_matching(const vector<vi>& g, int n, int m) {
-	match.assign(m, -1);
-	rep(i,0,n) {
-		seen.assign(m, 0);
+int dfsMatching(vector<vi> g, vi& btoa) {
+	rep(i,0,sz(g)) {
+		vector<bool> vis(sz(btoa));
```

Comment:
making this global again would remove an allocation here, and is likely shorter as well

Reply:
I don't think it saves an allocation - we still need to do `vis.assign(...)`. Main advantage would be that we don't need to pass it through the parameters.